src/cmdline/mod.rs

```diff
@@ -283,10 +194,11 @@ impl Cmdline {
     /// # use linux_loader::cmdline::*;
     /// # use std::ffi::CString;
     /// # use vm_memory::{GuestAddress, GuestUsize};
-    /// let mut cl = Cmdline::new(100);
-    /// cl.add_virtio_mmio_device(1 << 12, GuestAddress(0x1000), 5, Some(42))
+    /// let mut cl = Cmdline::new(100).unwrap();
+    /// cl.boot_args_mut()
+    ///     .add_virtio_mmio_device(1 << 12, GuestAddress(0x1000), 5, Some(42))
     ///     .unwrap();
-    /// let cl_cstring = CString::new(cl).unwrap();
+    /// let cl_cstring = CString::new(cl.as_str().unwrap()).unwrap();
```

Comment:
I don't think this needs to be a CString anymore.

Reply:
On a second thought, that where we get the null terminated string from. The reason why we're using CString here is that one, it was just very poorly documented.